Unstructured-IO/base-images
# OSS-35 Feat/push base images with hash in tag

###  Summary
 
This PR partially resolves https://unstructured-ai.atlassian.net/browse/OSS-35 by:

- add short hash to the base image tag so that new base images doesn't overwrite existing ones
- add new tags `wolfi-base-latest`, `rocky9.2-cpu-latest`, `rocky9.2-gpu-latest`,  `rocky9.2-slim-latest` that are the latest builds from ci; this is intended to facilitate smoother downstream development experience

### Test instructions

During test two tags are pushed to quay.io:
- `wolfi-base-latest`
- `wolfi-base-c62024e`

Try pull those locally  with arm64 platform they both should give the same image that has the hash `66f84447138b893b272e75044f9967379de3e874849cb2b9cb7aa2b3e9dea4e6`
if pull with platform being amd64 they should both have hash `7c3af225a39f730f4feee705df6cd8d1570739dc130456cf589ac53347da0f1d`

### Next steps

- add steps in ci to delete out of date images (either based on date or number of images in the repo with the same base)
- add steps in ci to automatically generate pull request to affect downstream repos to update image reference
